server-common/src/main/java/org/apache/gravitino/server/authorization/annotations/AuthorizeMetadata.java

```diff
@@ -0,0 +1,36 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *  http://www.apache.org/licenses/LICENSE-2.0
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+package org.apache.gravitino.server.authorization.annotations;
+
+import java.lang.annotation.ElementType;
+import java.lang.annotation.Retention;
+import java.lang.annotation.RetentionPolicy;
+import java.lang.annotation.Target;
+import org.apache.gravitino.MetadataObject;
+
+/** This annotation identify which parameters in the request are to be used for authorization. */
+@Target({ElementType.PARAMETER, ElementType.FIELD})
+@Retention(RetentionPolicy.RUNTIME)
+public @interface AuthorizeMetadata {
+  /**
+   * The name of the parameter to be used for authorization.
+   *
+   * @return the name of the parameter to be used for authorization.
+   */
+  MetadataObject.Type value();
```

Comment:
Why is it a type? Maybe we should give the interface a better name.

Reply:
He uses it to inform the interceptor about which type of metadata this parameter represents. You can get more information in issue #6828.

Reply:
Is it the suitable to use name `type`?

Reply:
Yes, I also think `type()` better `value()`.

Reply:
I'll remove them first and then complete it in PR #6828

Reply:
OK.